src/drivers/sdl/lv_sdl_window.c

```diff
@@ -209,8 +207,45 @@ static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_m
 #if LV_USE_DRAW_SDL == 0
     lv_sdl_window_t * dsc = lv_display_get_driver_data(disp);
     lv_color_format_t cf = lv_display_get_color_format(disp);
+    static uint32_t * argb_px_map = NULL;
 
     if(sdl_render_mode() == LV_DISPLAY_RENDER_MODE_PARTIAL) {
+
+        /*Update values in a special OLED I1 --> ARGB8888 case
+          We render everything in I1, but display it in ARGB8888*/
+        if(cf == LV_COLOR_FORMAT_I1) {
```

Comment:
Let's create a function from it [here](https://github.com/lvgl/lvgl/blob/master/src/draw/sw/lv_draw_sw.c#L220).
E.g. 
```c
void lv_draw_sw_i1_to_argb8888(void * buf_i1, void * buf_argb8888, size_t buf_size_px);
```